api/host.py

```diff
@@ -132,7 +135,53 @@ def get_host_list(
 @api_operation
 @rbac(Permission.WRITE)
 @metrics.api_request_time.time()
-def delete_by_id(host_id_list):
+def delete_host_list(
+    display_name=None,
+    fqdn=None,
+    hostname_or_id=None,
+    insights_id=None,
+    provider_id=None,
+    provider_type=None,
+    registered_with=None,
+    staleness=None,
+    tags=None,
+    filter=None,
+):
+    if not any([display_name, fqdn, hostname_or_id, insights_id, provider_id, provider_type, tags, filter]):
+        logger.error("bulk-delete operation needs at least one input property to filter on.")
+        flask.abort(400, "bulk-delete operation needs at least one input property to filter on.")
+
+    try:
+        ids_list = get_host_ids_list_xjoin(
+            display_name,
+            fqdn,
+            hostname_or_id,
+            insights_id,
+            provider_id,
+            provider_type,
+            registered_with,
+            staleness,
+            tags,
+            filter,
+        )
+    except ValueError as err:
+        log_get_host_list_failed(logger)
+        flask.abort(400, str(err))
+    except ConnectionError:
+        logger.error("xjoin-search not accessible")
+        flask.abort(503)
+
+    if not len(ids_list):
+        flask.abort(status.HTTP_404_NOT_FOUND)
+
+    delete_count = _delete_filtered_hosts(ids_list)
+
+    json_data = {"hosts_found": len(ids_list), "hosts_deleted": delete_count}
+
+    return flask_json_response(json_data, status.HTTP_202_ACCEPTED)
+
+
+def _delete_filtered_hosts(host_id_list):
```

Comment:
I think the name of the function is misleading since it's used in other places as well. Like when it's used as part of `delete_by_id` it's not deleting filtered hosts, it's just deleting hosts using a list of ids. I propose changing the name to reflect it's behavior and uses.
```suggestion
def _delete_hosts_by_id_list(host_id_list):
```